analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -0,0 +1,156 @@
+                var methodDeclaration = references.Single(x => x.Declaration.StartLine == 3);
+                methodDeclaration.Declaration.Should().BeEquivalentTo(new TextRange { StartLine = 3, EndLine = 3, StartOffset = 15, EndOffset = 21 });
+                methodDeclaration.Reference.Should().HaveCount(1);
+                methodDeclaration.Reference.Single().Should().BeEquivalentTo(new TextRange { StartLine = 7, EndLine = 7, StartOffset = 8, EndOffset = 14 });
+            });
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Event_CS() =>
+            Verify("Event.cs", 6, 5, 9, 10);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Field_CS() =>
+            Verify("Field.cs", 4, 3, 7, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Local_CS() =>
+            Verify("Local.cs", 4, 7, 5);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Method_CS() =>
+            Verify("Method.cs", 3, 3, 6);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_NamedType_CS() =>
+            Verify("NamedType.cs", 4, 3, 7, 7); // 'var' and type name on the same line
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Parameter_CS() =>
+            Verify("Parameter.cs", 4, 4, 6, 7);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Property_CS() =>
+            Verify("Property.cs", 4, 3, 7, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_Setter_CS() =>
+            Verify("Setter.cs", 4, 6, 8);
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Verify_TypeParameter_CS() =>
+            Verify("TypeParameter.cs", 5, 2, 4, 6);
+
+        public void Verify(string fileName, int expectedDeclarationCount, int assertedDeclarationLine, params int[] assertedDeclarationLineReferences) =>
+            Verify(fileName, references =>
+                {
+                    references.Where(x => x.Declaration != null).Should().HaveCount(expectedDeclarationCount);
+                    var declarationWithReferences = references.Single(x => x.Declaration.StartLine == assertedDeclarationLine).Reference;
+                    declarationWithReferences.Select(x => x.StartLine).Should().BeEquivalentTo(assertedDeclarationLineReferences);
+                });
+
+        public void Verify(string fileName, Action<IEnumerable<SymbolReferenceInfo.Types.SymbolReference>> verifyReference)
+        {
+            var testRoot = Root + TestContext.TestName;
+            UtilityAnalyzerBase analyzer = fileName.EndsWith(".cs")
+                ? new TestSymbolReferenceAnalyzer_CS(testRoot)
+                : new TestSymbolReferenceAnalyzer_VB(testRoot);
```

Comment:
We can extract the creation to a factory method.

Reply:
It's needed just once and easy to read with this formatting.